ccdproc/core.py

```diff
@@ -441,7 +441,13 @@ def create_deviation(ccd_data, gain=None, readnoise=None, disregard_nan=False):
         logging.warning("Negative values in array will be replaced with nan")
 
     # calculate the deviation
-    var = (xp.sqrt(data) ** 2 + readnoise_value**2) ** 0.5
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            "ignore",
+            message="invalid value encountered in sqrt",
+            category=RuntimeWarning,
+        )
+        var = (xp.sqrt(data) ** 2 + readnoise_value**2) ** 0.5
```

Comment:
```suggestion
        var = xp.sqrt(xp.abs(data) + readnoise_value**2)
```

Reply:
Turns out this is not equivalent in the case when `data` is negative -- then `sqrt` generates a `nan` which we want..